From: Modestas Vainius <modestas@vainius.eu>
Subject: [PATCH] debian/simple_top_makefile
Last-Update: 2011-04-23

Add a simple Makefile to the top level directory which
implements most README steps. Some stuff might be Debian
specific in here.

Signed-off-by: Modestas Vainius <modestas@vainius.eu>

---
 Makefile |   45 +++++++++++++++++++++++++++++++++++++++++++++
 1 files changed, 45 insertions(+), 0 deletions(-)

Index: qtscriptgenerator-src-0.2.0/Makefile
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ qtscriptgenerator-src-0.2.0/Makefile	2012-02-23 09:32:56.000000000 +0100
@@ -0,0 +1,48 @@
+BUILDDIR ?= .
+GENERATOR := $(BUILDDIR)/generator/generator
+CXXFLAGS ?=
+GENERATED_CPP_DONE = $(BUILDDIR)/generated_cpp/done
+QTBINDINGS_DONE = $(BUILDDIR)/qtbindings/done
+PREFIX ?= /usr/local
+
+##### Build generator and QtScript Bindings #####
+all: build
+
+$(BUILDDIR) $(BUILDDIR)/generator $(BUILDDIR)/qtbindings:
+	mkdir -p $@
+
+## Build generator ##
+$(BUILDDIR)/generator/Makefile: $(CURDIR)/generator/generator.pro | $(BUILDDIR)/generator
+	cd $(BUILDDIR)/generator && qmake QMAKE_CXXFLAGS="$(CXXFLAGS)" $(CURDIR)/generator/generator.pro
+
+$(GENERATOR): $(BUILDDIR)/generator/Makefile
+	$(MAKE) -C $(BUILDDIR)/generator $(QMAKE_TARGET)
+	touch $@
+
+## Generate bindings CPP files ##
+$(GENERATED_CPP_DONE): $(GENERATOR) $(wildcard generator/typesystem_*.xml)
+	cd generator && QTDIR=$(PREFIX) $(GENERATOR) --include-paths=$(PREFIX)/include --output-directory=$(BUILDDIR)
+	touch $@
+
+## Build qtbindings ##
+$(BUILDDIR)/qtbindings/Makefile: $(GENERATED_CPP_DONE) $(CURDIR)/qtbindings/qtbindings.pro | $(BUILDDIR)/qtbindings
+	cd $(BUILDDIR)/qtbindings && qmake QMAKE_CXXFLAGS="$(CXXFLAGS)" $(CURDIR)/qtbindings/qtbindings.pro
+
+$(QTBINDINGS_DONE): $(BUILDDIR)/qtbindings/Makefile
+	$(MAKE) -C $(BUILDDIR)/qtbindings
+	touch $@
+
+build: $(QTBINDINGS_DONE)
+
+clean distclean:
+	if [ -f "$(BUILDDIR)/qtbindings/Makefile" ]; then rm -f $(GENERATED_CPP_DONE); $(MAKE) -C $(BUILDDIR)/qtbindings $@; fi
+	if [ -f "$(BUILDDIR)/generator/Makefile" ]; then rm -f $(QTBINDINGS_DONE); $(MAKE) -C $(BUILDDIR)/generator $@; fi
+	rm -rf $(BUILDDIR)/generated_cpp $(BUILDDIR)/plugins
+	rm -f $(BUILDDIR)/doc/*.html
+	rm -f generator/.preprocessed.tmp generator/mjb_rejected_*.log
+
+install: build
+	$(MAKE) -C $(BUILDDIR)/generator install
+	$(MAKE) -C $(BUILDDIR)/qtbindings install
+
+.PHONY: clean distclean build install
